### DISCUSSION TITLE
Unable To create event using cookie

### DISCUSSION THREAD

**Discussion by mahajanmahesh935** [Sunbird-Ed/Community#726] [Bugs]
I am encountering an "Unauthorized" error when calling the api/event/v4/create API and passing the cookie in the headers. However, the request succeeds when using a Bearer token. Any assistance in resolving this issue would be greatly appreciated.

**CURL:**
curl --location '-------/api/event/v4/create' \
--header 'Accept: application/json, text/plain, */*' \
--header 'Accept-Language: en-US,en;q=0.9' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--header 'Cookie: ------- \
--header 'Origin: --------' \
--header 'Referer: ---------/workspace/add/event' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36' \
--header 'sec-ch-ua: "Google Chrome";v="125", "Chromium";v="125", "Not.A/Brand";v="24"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "Linux"' \
--data '{
    "request": {
        "event": {
            "name": "DemoEvent",
            "eventType": "Online",
            "description": "description",
            "code": "11",
            "onlineProvider": "Google Meet",
            "eventVisibility": "Public",
            "language": [
                "English"
            ],
            "source": "sdasd",
            "startDate": "2024-06-07",
            "startTime": "15:30:10+05:30",
            "endDate": "2024-06-21",
            "endTime": "16:30:10+05:30",
            "registrationStartDate": "2024-06-07",
            "registrationEndDate": "2024-06-08",
            "onlineProviderData": {},
            "venue": {},

        }
    }
}

**response:**
{
    "message": "Unauthorized"
}

@rajeevsathish @AmiableAnil @rajeevsathish

**Comment by mahajanmahesh935** [comment]
Hi team,
Your insights and assistance would be highly appreciated in resolving this issue. Looking forward to your response.

Thank you.

**Reply by vinukumar-vs** [reply]
@mahajanmahesh935 Can you share more details 
* Which version of sunbrid are you using?
* Are you using sunbirdEd-portal for UI & proxy or you have built your own portal application?

If you are using sunbirdEd-portal, verify the end-point is whitelisted or not and also verify the role permission given to access this end-point.

https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-5.2.0/src/app/helpers/whitelistApis.js
'/event/v4/create'

Also i have notices, you are calling the kong end-point directly. `{{host}}**/api/**event/v4/create'`. I would suggest you to write proxy end-point on portal-backend.
@rajeevsathish @gandham-santhosh

**Comment by mahajanmahesh935** [comment]
Hi @vinukumar-vs 
We are using release-4.10_RC12 and Using sunbirdED-portal for UI and proxy.
Also addded the "/event/v4/create"  and 
```
const API_LIST = {
  URL: {
"/event/v4/create":{
      checksNeeded: ["ROLE_CHECK"],
      ROLE_CHECK: [
        ROLE.CONTENT_CREATOR
      ],
    }
}
}
```
this end point in whitelistApis.js but still getting the Unauthorized issue

**Reply by rajeevsathish** [reply]
is this the new API you have introduced @mahajanmahesh935  and if yer have you added it in the RBAC?
